modules/humansecurityRtdProvider.js

```diff
@@ -0,0 +1,179 @@
+    throw new Error(`The 'clientId' parameter must be a short alphanumeric string`);
+  }
+
+  // Load/reset the state
+  verbose = !!config?.params?.verbose;
+  sessionId = generateUUID();
+  hmnsData = {};
+
+  // We rely on prebid implementation to get the best domain possible here
+  // In some cases, it still might be null, though
+  const refDomain = getRefererInfo().domain || '';
+
+  // Once loaded, the implementation script will publish an API using
+  // the session ID value it was given in data attributes
+  const scriptAttrs = { 'data-sid': sessionId };
+  const scriptUrl = `${SCRIPT_URL}?r=${refDomain}${clientId ? `&c=${clientId}` : ''}`;
+
+  loadExternalScript(scriptUrl, SUBMODULE_NAME, onImplLoaded, null, scriptAttrs);
+}
+
+/**
+ * The callback to loadExternalScript
+ * Establishes the bridge between this RTD submodule and the loaded implementation
+ */
+function onImplLoaded() {
+  // We then get a hold on this script using the knowledge of this session ID
+  const wnd = getWindowSelf();
+  const impl = wnd[`sonar_${sessionId}`];
+  if (typeof impl !== 'object' || typeof impl.connect !== 'function') {
+    verbose && logWarn('onload', 'Unable to access the implementation script');
+    return;
+  }
+
+  // And set up a bridge between the RTD submodule and the implementation.
+  // The first argument is used to identify the caller.
+  // The callback might be called multiple times to update the signals
+  // once more precise information is available.
+  impl.connect(getGlobal(), onImplMessage);
+}
+
+/**
+ * The bridge function will be called by the implementation script
+ * to update the token information or report errors
+ * @param {Object} msg
+ */
+function onImplMessage(msg) {
+  if (typeof msg !== 'object') {
+    return;
+  }
+
+  switch (msg.type) {
+    case 'hmns': {
+      hmnsData = mergeDeep({}, msg.data || {});
+      break;
+    }
+    case 'error': {
+      logError('impl', msg.data || '');
+      break;
+    }
+    case 'warn': {
+      verbose && logWarn('impl', msg.data || '');
+      break;
+    }
+    case 'info': {
+      verbose && logInfo('impl', msg.data || '');
+      break;
+    }
+  }
+}
+
+/**
+ * onGetBidRequestData is called once per auction.
+ * Update the `ortb2Fragments` object with the data from the injected script.
+ *
+ * @param {Object} reqBidsConfigObj
+ * @param {function} callback
+ * @param {SubmoduleConfig} config
+ * @param {UserConsentData} userConsent
+ */
+function onGetBidRequestData(reqBidsConfigObj, callback, config, userConsent) {
+  // Add device.ext.hmns to the global ORTB data for all vendors to use
+  // At the time of writing this submodule, "hmns" is an object defined
+  // internally by humansecurity, and it currently contains "v1" field
+  // with a token that contains collected signals about this session.
+  mergeDeep(reqBidsConfigObj.ortb2Fragments.global, { device: { ext: { hmns: hmnsData } } });
```

Comment:
I think this is the right place; you may wish to chat with @jwrosewell about his efforts to get everyone absorbing it